From 99d099757ab487e0d317c69541b47aaae0b6c431 Mon Sep 17 00:00:00 2001
Message-Id: <99d099757ab487e0d317c69541b47aaae0b6c431.1650799193.git.geliang.tang@suse.com>
From: Jiri Olsa <jolsa@redhat.com>
Date: Thu, 21 Oct 2021 13:41:32 +0200
Subject: [PATCH] selftests/bpf: Use nanosleep tracepoint in perf buffer test
Git-commit: 99d099757ab487e0d317c69541b47aaae0b6c431
Patch-mainline: v5.16-rc1
References: bsn#127
Modified-by-SEL: No


The perf buffer tests triggers trace with nanosleep syscall,
but monitors all syscalls, which results in lot of data in the
buffer and makes it harder to debug. Let's lower the trace
traffic and monitor just nanosleep syscall.

Signed-off-by: Jiri Olsa <jolsa@kernel.org>
Signed-off-by: Andrii Nakryiko <andrii@kernel.org>
Acked-by: John Fastabend <john.fastabend@gmail.com>
Link: https://lore.kernel.org/bpf/20211021114132.8196-4-jolsa@kernel.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 tools/testing/selftests/bpf/progs/test_perf_buffer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/testing/selftests/bpf/progs/test_perf_buffer.c b/tools/testing/selftests/bpf/progs/test_perf_buffer.c
index d37ce29fd393..a08874c5bdf2 100644
--- a/tools/testing/selftests/bpf/progs/test_perf_buffer.c
+++ b/tools/testing/selftests/bpf/progs/test_perf_buffer.c
@@ -12,7 +12,7 @@ struct {
 	__type(value, int);
 } perf_buf_map SEC(".maps");
 
-SEC("tp/raw_syscalls/sys_enter")
+SEC("tp/syscalls/sys_enter_nanosleep")
 int handle_sys_enter(void *ctx)
 {
 	int cpu = bpf_get_smp_processor_id();
-- 
2.26.2

